azkaban-common/src/main/java/azkaban/event/Event.java

```diff
@@ -53,24 +53,13 @@ public long getTime() {
     return time;
   }
 
-  public Object getData() {
+  public EventData getData() {
     return eventData;
   }
 
-  public static Event create(Object runner, Type type) {
-    return new Event(runner, type, null, true);
-  }
-
-  public static Event create(Object runner, Type type, Object eventData) {
-    return new Event(runner, type, eventData, true);
+  public static Event create(Object runner, Type type, EventData eventData) {
+    Preconditions.checkNotNull(eventData, "EventData was null");
```

Comment:
This will throw an exception if the condition is not met. How did you make sure that all the callers don't pass in null? 

It's a risk that the exception may cause a thread to die. Before this change, such a bug may have been hidden but may not cause a production issue.